### PR TITLE
Update gamepad for new hardware

### PR DESCRIPTION
**Pull request by Brycey92**
### Description

Update gamepad UI, touch and accel backend, and accel menu setting
### Test Instructions

Tested on emulator by me, tested by @jt-moriarty on hardware

### Readiness Checklist
- [x] I have run `make format` to format the changes
- [x] I have compiled the firmware and the changes have no warnings
- [x] I have compiled the emulator and the changes have no warnings
- [x] I have run `make cppcheck` and checked that `cppcheck_result.txt` has no warnings for the changes
- [x] I have added doxygen comments to any code used by more than one Swadge mode. This includes `/*! \file` comments with Design Philosophy, Usage, and Example sections for new headers.
- [x] I have run `make docs` and checked that `doxy_warnings.txt` has no warnings for the new code

### CODE REVIEW

**Comment by AEFeinstein** [comment]
This is a two-way lie. The button names don't match silk, and also it doesn't work (yet)
```c
const char reminderText[] = "Start + Select to Exit";
```

File: main/modes/gamepad/gamepad.c [hunk 1]
```diff
@@ -942,6 +832,7 @@ void gamepadReportStateToHost(void)
             }
             case GAMEPAD_NS:
             {
+                // TODO: accel
                 // TODO check this
```
Comment (AEFeinstein) [inline main/modes/gamepad/gamepad.c:836]: You should probably do this 😜

**Review by AEFeinstein — CHANGES_REQUESTED** [review]
Polish up the exit path, check Switch accel, and this'll be good to go!

**Comment by Brycey92** [comment]
I don't have a Switch here in NY, so Switch accel will have to fall to @JVeg199X

File: main/modes/gamepad/gamepad.c [hunk 2]
```diff
@@ -340,6 +344,12 @@ void gamepadStart(gamepadType_t type)
     }
     else
     {
+        const uint8_t macBytes = 6; // This is part of the ESP API's design, and cannot be changed here
+        uint8_t mac[macBytes];
+        if (ESP_OK == esp_wifi_get_mac(WIFI_IF_STA, mac))
+        {
+            memcpy(&hid_string_descriptor[3], mac, macBytes);
+        }
```
Comment (AEFeinstein) [inline main/modes/gamepad/gamepad.c:352]: Is the serial expecting alphanumeric or ASCII values? because the MAC is going to be over the place.

Comment (AEFeinstein) [reply main/modes/gamepad/gamepad.c:352]: And the string being overwritten is "123456" (which is also the combination to my luggage.

File: main/modes/gamepad/gamepad.c [hunk 3]
```diff
@@ -703,6 +725,18 @@ void gamepadButtonCb(buttonEvt_t* evt)
...
+                if ((evt->state & PB_START) && (evt->state & PB_SELECT))
+                {
+                    gamepad->exitTimer = 1;
+                }
+                else
+                {
+                    gamepad->exitTimer = 0;
+                }
+            }
+
```
Comment (AEFeinstein) [inline main/modes/gamepad/gamepad.c:739]: No exit for Switch?

File: main/swadge2024.c [hunk 4]
```diff
@@ -157,7 +157,6 @@
     #define RTC_DATA_ATTR
 #endif
 
-#define EXIT_TIME_US          1000000
 #define DEFAULT_FRAME_RATE_US 40000
```
Comment (AEFeinstein) [inline main/swadge2024.c:160]: Hah, I moved {{DEFAULT_FRAME_RATE_US}} to the header too. You're gonna want to `git merge origin/main` here.

Comment (Brycey92) [reply main/swadge2024.c]: Heh

File: main/modes/gamepad/gamepad.c (hunk 3 above)
Comment (Brycey92) [reply main/modes/gamepad/gamepad.c]: Exit'd

File: main/modes/gamepad/gamepad.c (hunk 2 above)
Comment (AEFeinstein) [reply main/modes/gamepad/gamepad.c:351]: @Brycey92 to check if there are any collisions with two swadges with the same serial. If not, remove this.

Comment (Brycey92) [reply main/modes/gamepad/gamepad.c]: Removed, no conflicts between 2 swadges

**Review by AEFeinstein — APPROVED** [review]
This looks and feels good, so I'm gonna merge it.
@JVeg199X I believe it's on you to test some NS stuff. If changes come out of that, they can be handled in a separate PR.

**Pull request merged**